model/remote/list.go

```diff
@@ -0,0 +1,72 @@
+package remote
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cozy/cozy-stack/model/instance"
+	"github.com/cozy/cozy-stack/pkg/config/config"
+	"github.com/cozy/httpcache"
+)
+
+var listClient = &http.Client{
+	Timeout:   20 * time.Second,
+	Transport: httpcache.NewMemoryCacheTransport(32),
+}
+
+// https://docs.github.com/en/rest/repos/contents#get-repository-content
+const listURL = "https://api.github.com/repos/cozy/cozy-doctypes/contents/"
+
+type listEntries struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// ListDoctypes returns the list of the known remote doctypes.
+func ListDoctypes(inst *instance.Instance) ([]string, error) {
+	var doctypes []string
+	if config.GetConfig().Doctypes == "" {
+		req, err := http.NewRequest(http.MethodGet, listURL, nil)
+		if err != nil {
+			return nil, err
+		}
+		res, err := listClient.Do(req)
+		if err != nil {
+			log.Warnf("cannot list doctypes: %s", err)
+			return nil, ErrNotFoundRemote
+		}
+		defer res.Body.Close()
+		var entries []listEntries
+		if err := json.NewDecoder(res.Body).Decode(&entries); err != nil {
+			return nil, err
+		}
+		for _, entry := range entries {
+			if entry.Type != "dir" {
+				continue
+			}
+			if entry.Name == "docs" || strings.HasPrefix(entry.Name, ".") {
+				continue
+			}
+			doctypes = append(doctypes, entry.Name)
+		}
+	} else {
+		entries, err := ioutil.ReadDir(config.GetConfig().Doctypes)
+		if err != nil {
+			return nil, err
+		}
+		for _, entry := range entries {
+			if !entry.IsDir() {
+				continue
+			}
+			name := entry.Name()
+			if name == "docs" || strings.HasPrefix(name, ".") {
+				continue
+			}
+			doctypes = append(doctypes, name)
+		}
+	}
```

Comment:
We'll have to keep this in mind if we ever change the cozy-doctypes repo organization

Reply:
Yes, but it was already the case before this PR. If the organization of cozy-doctypes repo is changed, the stack won't find the remote doctypes definition.